internal/db/transaction/manager.go

```diff
@@ -226,19 +226,13 @@ func unmarshalTransactionReceipt(b []byte) (types.TxnReceipt, error) {
 		return nil, err
 	}
 	if receipt := protoReceipt.GetInvoke(); receipt != nil {
-		var messagesSent []*types.MsgToL1 = nil
-		if len(receipt.MessagesSent) > 0 {
-			messagesSent = make([]*types.MsgToL1, len(receipt.MessagesSent))
-			for i, msg := range receipt.MessagesSent {
-				messagesSent[i] = unmarshalMessageL2ToL1(msg)
-			}
+		messagesSent := make([]*types.MsgToL1, len(receipt.MessagesSent))
+		for i, msg := range receipt.MessagesSent {
+			messagesSent[i] = unmarshalMessageL2ToL1(msg)
```

Comment:
It might be more efficient to avoid zeroing out the array as follows (plus we avoid tracking an iteration variable):

```suggestion
		messagesSent := make([]*types.MsgToL1, 0, len(receipt.MessagesSent))
		for _, msg := range receipt.MessagesSent {
			messagesSent = append(messagesSent, unmarshalMessageL2ToL1(msg))
```